test/e2e_federation/util.go

```diff
@@ -150,20 +139,85 @@ func createService(clientset *fedclientset.Clientset, namespace, name string) (*
 	return clientset.Services(namespace).Create(service)
 }
 
-func createServiceOrFail(clientset *fedclientset.Clientset, namespace, name string) *v1.Service {
-	service, err := createService(clientset, namespace, name)
+func createService(clientset *fedclientset.Clientset, namespace, name string, clusters fedframework.ClusterSlice) (*v1.Service, error) {
+	if clientset == nil || len(namespace) == 0 {
+		return nil, fmt.Errorf("Internal error: invalid parameters passed to createService: clientset: %v, namespace: %v", clientset, namespace)
+	}
+	By(fmt.Sprintf("Creating federated service %q in namespace %q", name, namespace))
+
+	// Tests can be run in parallel, so we need a different nodePort for
+	// each test.
+	// we add in a array all the "available" ports
+	availablePorts := make([]int32, FederatedSvcNodePortLast-FederatedSvcNodePortFirst)
+	for i := range availablePorts {
+		availablePorts[i] = int32(FederatedSvcNodePortFirst + i)
```

Comment:
This list of available ports is kind of very big. in case of some network error to some cluster. we would keep on retrying forever in the loop. So i would like it to limit to 5-10 iterations at max.